publish-mqtt/src/main.rs

```diff
@@ -90,29 +69,48 @@ async fn main() -> Result<(), Box<dyn Error + Send + Sync>> {
 
     let local = task::LocalSet::new();
 
-    let bluetooth_handle = local.spawn_local(async move {
-        bluetooth_mainloop(homie).await.unwrap();
-    });
+    // Connect a bluetooth session.
+    let (dbus_handle, bt_session) = MijiaSession::new().await?;
+
+    let bluetooth_handle =
+        local.run_until(async move { bluetooth_mainloop(homie, bt_session).await });
 
     // Poll everything to completion, until the first one bombs out.
-    let res: Result<_, Box<dyn Error + Send + Sync>> = try_join! {
-        // LocalSet finished first. Colour me confused.
-        local.map(|()| Ok(println!("WTF?"))),
+    let res: Result<_, anyhow::Error> = try_join! {
+        // The resource is a task that should be spawned onto a tokio compatible
+        // reactor ASAP. If the resource ever finishes, you lost connection to D-Bus.
+        dbus_handle,
         // Bluetooth finished first. Convert error and get on with your life.
         bluetooth_handle.map(|res| Ok(res?)),
         // MQTT event loop finished first.
-        mqtt_handle,
+        mqtt_handle.map(|res| Ok(res.map_err(|err| anyhow::anyhow!(err))?)),
```

Comment:
Why do you need to do the `Ok(...?)` thing here? Isn't the error type already converted appropriately by the `map_err`?

Reply:
I probably don't